### ISSUE TITLE
linter: react/no-unstable-nested-components matches propNamePattern against the JSX attribute name instead of the object property key

### ISSUE THREAD

**Issue by fant0m121** [oxc-project/oxc#26086]
### What version of Rust are you using?

n/a (using the published `oxlint` npm binary)

### What version of oxlint are you using?

1.78.0

### What command did you run?

```sh
oxlint -c oxlint.config.json repro.tsx
```

### Config

`oxlint.config.json`:

```json
{
  "plugins": ["react"],
  "categories": {},
  "settings": { "react": { "version": "19.0" } },
  "rules": {
    "react/no-unstable-nested-components": ["error", { "propNamePattern": "render*" }]
  }
}
```

### What did you expect to see?

`propNamePattern` should be matched against the name of the object property that actually holds the function, not against the name of the JSX attribute that the object literal happens to be nested in.

`repro.tsx`:

```tsx
function DataGrid(_props: { columns: unknown[] }) {
  return null;
}

// A: array extracted to a variable
export function Extracted() {
  const columns = [{ name: "a", render: ({ id }: { id: string }) => <div>{id}</div> }];

  return <DataGrid columns={columns} />;
}

// B: the same array inlined into the JSX attribute
export function Inlined() {
  return <DataGrid columns={[{ name: "a", render: ({ id }: { id: string }) => <div>{id}</div> }]} />;
}
```

Both `Extracted` and `Inlined` put the component under a `render` object key, so with `propNamePattern: "render*"` neither should be reported.

This is also what `eslint-plugin-react` does. Its `isDirectValueOfRenderProperty` helper matches the pattern against `node.parent.key.name` (the object property key) and its docblock names the inlined form explicitly as a supported case:

```js
/**
 * Check whether given node is declared directly inside a render property
 * ```jsx
 * const rows = { render: () => <div /> }
 * <Component rows={ [{ render: () => <div /> }] } />
 *  ```
 */
function isDirectValueOfRenderProperty(node, propNamePattern) {
  return (
    node
    && node.parent
    && node.parent.type === 'Property'
    && node.parent.key
    && node.parent.key.type === 'Identifier'
    && propMatchesRenderPropPattern(node.parent.key.name, propNamePattern)
  );
}
```

(`eslint-plugin-react@7.37.5`, `lib/rules/no-unstable-nested-components.js`)

### What did you see instead?

Only case B is reported. The pattern appears to be matched against the JSX attribute name `columns` instead of the object key `render`, so semantically identical code behaves differently depending on whether the array is extracted into a variable:

```
repro.tsx:14:51: error react(no-unstable-nested-components): Do not define components during render.
  help: Move this component definition out of the parent component `Inlined` and pass data as props.
        If you want to allow component creation in props, set `allowAsProps` option to true.
```

No combination of `propNamePattern` values fixes this: the only pattern that silences case B is one that literally contains `columns`, which means "allow declaring components in any `columns` prop" — much weaker and less obvious than the intended `render*`.

### Impact

In our monorepo the inlined form is the dominant one: of 107 reports from this rule in one app, 86 are case B — data-grid column definitions inlined into a `columns` attribute. Because `propNamePattern` cannot express them, the only workable option today is `allowAsProps: true`, which also disables the checks we do want to keep for other props.

I checked the changelog on `main` (i.e. including 1.79 and 1.80): the only entries for this rule are the initial implementation (#22248) and the schema addition (#23287), both before 1.74, so this does not look like something already fixed upstream.
